packages/istanbul-reports/lib/clover/index.js

```diff
@@ -135,13 +135,35 @@ class CloverReport extends ReportBase {
                 count,
                 type: 'stmt'
             };
-            const branchDetail = branchByLine[k];
+            const branchDetail = branchDetails[k];
 
-            if (branchDetail) {
-                attrs.type = 'cond';
-                attrs.truecount = branchDetail.covered;
-                attrs.falsecount = branchDetail.total - branchDetail.covered;
+            if (!branchDetail || branchDetail.type === 'switch') {
+                return this.xml.inlineTag('line', attrs);
             }
+
+            attrs.type = 'cond';
+            attrs.truecount = 0;
+            attrs.falsecount = 0;
+
+            if (count === 0) {
+                return this.xml.inlineTag('line', attrs);
+            }
+
+            if (['if', 'cond-expr'].includes(branchDetail.type)) {
+                attrs.truecount = branchDetail.states[0];
+                attrs.falsecount = branchDetail.states[1];
+            } else if (
+                ['binary-expr', 'default-arg'].includes(branchDetail.type)
+            ) {
+                if (branchDetail.states.every(state => state > 0)) {
+                    attrs.truecount = 1;
+                    attrs.falsecount = 1;
+                } else {
+                    attrs.truecount = 1;
+                    attrs.falsecount = 0;
```

Comment:
Hi, @bcoe. The problem mentioned in issue #695 looks to make others puzzled now so I think it's needed to push this PR.

I guess this PR stop forward because you may disagree with the practice that assigning hardcode values to make 3rd-party parsers understand if this line was covered or not.

Could we have better practice to resolve this problem?